sdk/core/core-client-rest/src/common.ts

```diff
@@ -215,6 +215,12 @@ export type ClientOptions = PipelineOptions & {
    * Specify a custom HttpClient when making requests.
    */
   httpClient?: HttpClient;
+  /**
+   * According to the guideline we should always use the client api-version - https://azure.github.io/azure-sdk/general_design.html#general-service-apiversion-7
+   * However some services will be broken if we override the existing api-version
+   * For backward compatibility we provide an option for them to ensure their call could be handled correctly
+   */
+  keepApiVersionInUrl?: boolean;
```

Comment:
@joheredi After confirming with ContainerService team currently the returned old apiVersion is by design and they also want to fix it but don't know when.

To not break our release [pr](https://github.com/Azure/azure-sdk-for-js/pull/22417) which is asked from azure portal team we prefer we add an option `keepApiVersionInUrl` here to ensure the package could work e.g ContainService.

Jose, how do you think?

/cc @lirenhe @qiaozha

Reply:
I don't love the idea of adding this option. I would rather add a custom apiVersion policy in the ContainerService library and remove it when the bug gets fixed

Reply:
Thanks for your suggestion! I'll have a try in containerservice pr.